src/client/utils/browser.js

```diff
@@ -1,48 +1,50 @@
-function getMSEdgeVersion (userAgent) {
-    var edgeStrIndex = userAgent.indexOf('edge/');
-
-    return parseInt(userAgent.substring(edgeStrIndex + 5, userAgent.indexOf('.', edgeStrIndex)), 10);
-}
-
 function calculateBrowserAndVersion (userAgent) {
-    var webkitRegEx  = /(webkit)[ \/]([\w.]+)/;
-    var msieRegEx    = /(msie) ([\w.]+)/;
-    var firefoxRegEx = /(firefox)/;
-
-    var match = webkitRegEx.exec(userAgent) ||
-                msieRegEx.exec(userAgent) ||
-                userAgent.indexOf('compatible') < 0 && firefoxRegEx.exec(userAgent) ||
-                [];
+    var msEdgeRegEx   = /(edge)\/([\w.]+)/;
+    var chromeRegEx   = /(chrome)[\/]([\w.]+)/;
+    var safariRegEx   = /(webkit)[ \/]([\w.]+).*(version)[ \/]([\w.]+).*(safari)[ \/]([\w.]+)/;
+    var webkitRegEx   = /(webkit)[\/]([\w.]+)/;
+    var ie9And10RegEx = /(msie) ([\w.]+)/;
+    var ie11RegEx     = /(rv)(?::| )([\w.]+)/;
+    var firefoxRegEx  = /(mozilla)(?:.*? rv:([\w.]+)|)/;
+    var match         = msEdgeRegEx.exec(userAgent) ||
+                        chromeRegEx.exec(userAgent) ||
+                        safariRegEx.exec(userAgent) ||
+                        webkitRegEx.exec(userAgent) ||
+                        ie9And10RegEx.exec(userAgent) ||
+                        userAgent.indexOf('trident') >= 0 && ie11RegEx.exec(userAgent) ||
+                        userAgent.indexOf('compatible') < 0 && firefoxRegEx.exec(userAgent) ||
+                        [];
 
     return {
-        name:    match[1] || '',
-        version: match[2] || '0'
+        name:    match[5] || match[3] || match[1] || '',
+        version: parseInt(match[4] || match[2] || '0', 10)
```

Comment:
If it is well-known browser then name and version will calcualte at once.
It it is browser based on popular engine (e.g. Yander browser) we will try calculate name and version, then engine name and version etc.